content/movie-review-pt2.md

```diff
@@ -0,0 +1,522 @@
+# Movie Review Program Pt2
+
+### List links for further reading here:
+
+- 
+
+# Lesson Objectives
+
+- Derive a Program Derived Address (PDA)
+- Invoke the System Program to create a new account using a PDA
+- Serialize data to the PDA account
+
+# TL;DR
+
+- Creating an account requires that we calculate the space and rent to allocate for a new account
+- A Program Derived Address (PDA) is derived from a program ID and an optional list of seeds
+- Creating a new account requires that perform a Cross Program Invocation (CPI) to the `create_account` instruction on the System Program
+- Updating the data field on an account requires that we serialize (convert to byte array) the data into the account
+
+# Overview
+
+In the previous lesson we went over how to, deserialize instruction data, deserialize account data, and print messages to the program log.
+
+In this lesson we will build upon what we have already done. The next step is to create a new account to store the movie review data from the user.
+
+## Account State
+
+In order to store data on a new account, we must first define a struct that serves as the format for the data field of the account. These structs are stored in a `state.rs` file and are used to serialize data into an account and deserialize data from an existing account.
+
+For our movie review program, we will store the rating, description, and title provided by the user. We will also want a field to check whether the account storing the review has already been initialized. This is to prevent duplicate reviews of the same movie by the same user. We’ll refer to this format as the `MovieAccountState`.
+
+To store data on an account, we must serialize the data. Serialization is the process of converting an object into a bytes array (an area of memory containing a continuous sequence of bytes). For our use case, the object is simply the data we want to store. Writing to the data field of an account requires that we first serialization the data we want to store.
+
+Deserialization is the process of reconstructing an object from a byte array. To read the data from an existing account, we must deserialized the account data from bytes into readable objects. We can then update this data and reserialize the data into the account.
```

Comment:
@ZYJLiu  line34 - do we mean "reserialize" as in re-serialize, i.e. to serialize again what was once serialized, then de-serialized

...or is this a typo... for serialize (or deserialize)?

Reply:
updated sentence to 

>  We can then update this (deserialized) data and serialize the data into the account again.

meant to say "re-serialize, i.e. to serialize again what was once serialized"